File: backend/src/main/java/com/runway/course/dto/ParticipatedCourseItem.java
package com.runway.course.dto;

import lombok.Builder;
import lombok.Getter;

import java.time.Instant;
import java.util.UUID;

@Getter
@Builder
public class ParticipatedCourseItem {

    private UUID courseId;
    private String name;
    private String description;
    private Double distanceMeters;
    private Boolean isLoop;
    private String status;
    private Integer attemptCountByMe;
    private Integer completionCountByMe;
    private Integer bestTimeSecondsByMe;
    private Instant lastAttemptAt;
}

File: android/app/src/main/java/com/runway/android/data/course/model/ParticipatedCourseItem.kt
package com.runway.android.data.course.model

data class ParticipatedCourseItem(
    val courseId: String,
    val name: String,
    val description: String?,
    val distanceMeters: Double,
    val isLoop: Boolean,
    val status: String,
    val attemptCountByMe: Int,
    val completionCountByMe: Int,
    val bestTimeSecondsByMe: Int?,
    val lastAttemptAt: String?,
)
